### PR TITLE
[fix](memory) fix invalid large memory check && fix memory info thread safety

### PR DESCRIPTION
**Pull request by xinyiZzz**
## Proposed changes

1. fix invalid large memory check
2. fix memory info thread safety

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by xinyiZzz** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.27 seconds
 stream load tsv:          511 seconds loaded 74807831229 Bytes, about 139 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          66 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.5 seconds inserted 10000000 Rows, about 338K ops/s
 storage size: 17158560463 Bytes

**Comment by xinyiZzz** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.08 seconds
 stream load tsv:          511 seconds loaded 74807831229 Bytes, about 139 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          66 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.2 seconds inserted 10000000 Rows, about 342K ops/s
 storage size: 17162306303 Bytes

File: be/src/runtime/memory/thread_mem_tracker_mgr.h [hunk 1]
```diff
@@ -173,10 +173,13 @@ inline void ThreadMemTrackerMgr::consume(int64_t size) {
     }
     // Large memory alloc should use allocator.h
     // Direct malloc or new large memory, unable to catch std::bad_alloc, BE may OOM.
-    if (size > 1024l * 1024 * 1024 && !doris::config::disable_memory_gc) { // 1G
+    if (large_memory_check && size > 1024l * 1024 * 1024 &&
```
Comment (yiguolei) [inline be/src/runtime/memory/thread_mem_tracker_mgr.h:176]: 把这个1G 改为config 中的参数吧

File: be/src/runtime/thread_context.h [hunk 2]
```diff
@@ -86,11 +86,11 @@
 #define SCOPED_TRACK_MEMORY_TO_UNKNOWN() (void)0
 #endif
 
-#define SKIP_MEMORY_CHECK(...)                  \
-    do {                                        \
-        doris::skip_memory_check++;             \
```
Comment (yiguolei) [inline be/src/runtime/thread_context.h:91]: should remove this thread local variable?

File: be/src/vec/common/allocator.h [hunk 3]
```diff
@@ -95,6 +96,8 @@ class Allocator {
     /// Allocate memory range.
     void* alloc(size_t size, size_t alignment = 0) {
         memory_check(size);
+        doris::thread_context()->large_memory_check = false;
```
Comment (yiguolei) [inline be/src/vec/common/allocator.h:99]: 为啥走allocator的就不检查large memory check了啊？

File: be/src/runtime/memory/thread_mem_tracker_mgr.h (hunk 1 above)
Comment (xinyiZzz) [reply be/src/runtime/memory/thread_mem_tracker_mgr.h:176]: 好

File: be/src/runtime/thread_context.h (hunk 2 above)
Comment (xinyiZzz) [reply be/src/runtime/thread_context.h:91]: memtable flush那里用到了，跳过memory check，因为不希望flush过程中失败

另外这个宏debug的时候也有些用

File: be/src/vec/common/allocator.h (hunk 3 above)
Comment (xinyiZzz) [reply be/src/vec/common/allocator.h:99]: 因为 Allocator 申请大内存是符合预期的，

Allocator之外申请大内存不符合预期，如果发现了，期望能改用Allocator申请，否则内存超限后不受控制的大内存申请容易导致OOM

**Comment by xinyiZzz** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]
LGTM

**Comment by xinyiZzz** [comment]
run buildall

**Review by yiguolei — DISMISSED** [review]
LGTM

**Comment by yiguolei** [comment]
run buildall

**Comment by xinyiZzz** [comment]
run buildall

**Comment by xinyiZzz** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 47.46 seconds
 stream load tsv:          511 seconds loaded 74807831229 Bytes, about 139 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.5 seconds inserted 10000000 Rows, about 338K ops/s
 storage size: 17156079660 Bytes

**Comment by xinyiZzz** [comment]
run buildall

**Comment by xinyiZzz** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.08 seconds
 stream load tsv:          503 seconds loaded 74807831229 Bytes, about 141 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          66 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17169563357 Bytes

**Review by yiguolei — APPROVED** [review]
LGTM

**Review by wangbo — APPROVED** [review]
LGTM

**Pull request merged**